icu4c/source/common/lsr.h

```diff
@@ -0,0 +1,66 @@
+// © 2019 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html#License
+
+// lsr.h
+// created: 2019may08 Markus W. Scherer
+
+#ifndef __LSR_H__
+#define __LSR_H__
+
+#include "unicode/utypes.h"
+#include "unicode/uobject.h"
+#include "cstring.h"
+
+U_NAMESPACE_BEGIN
+
+struct LSR final : public UMemory {
+    static constexpr int32_t REGION_INDEX_LIMIT = 1001 + 26 * 26;
+
+    const char *language;
+    const char *script;
+    const char *region;
+    char *owned;
+    int32_t languageLength;
+    int32_t scriptLength;
```

Comment:
I started with just pointers, then found that I sometimes needed the lengths, so I added them as fields. As discussed, I ended up using the lengths only during the hash code computation, so I am changing the code to compute & store the hash code for some LSRs, and remove the lengths again.

Otherwise, I could have used StringPiece. Alignment and struct size are not critical for an LSR.